mitodl/open-discussions
# Begin sharing (some) frontend code + styling between open-discussions and infinite-corridor

#### Pre-Flight checklist


#### What are the relevant tickets?
Towards https://github.com/mitodl/hq/discussions/95

#### What's this PR do?
This PR has very little new functionality, but sets up patterns/infrastructure to share code and styling between `open-discussions` and `infinite-corridor`. The frontends directory now has four yarn workspaces (local node packages):

```
frontends/
|-- open-discussions
|-- infinite-corridor
|-- ol-forms                ... Generic form-related things
|-- ol-util                 ... Very generic utilities
```

Infinite corridor still does not have much in it.

**Additionally:** This adds an env variable (defaulting to false) that enables/disables infinite_corridor site.

#### How should this be manually tested?
1. Restart docker *This will take a while the first time. You might want to look at the `watch` logs:*

    ```
    docker compose logs --follow --no-log-prefix  watch
    ````
2. Check new `ENABLE_INFINITE_CORRIDOR` env variable:
    1. Visit http://yourname.od.odl.local:8063/infinite ... expect 404
    2. Set `ENABLE_INFINITE_CORRIDOR=True` in env file and `docker compose restart watch`
    4. Visit http://yourname.od.odl.local:8063/infinite ... should work
3. Check open-discussions form errors:
    1. Go to  http://cac.od.odl.local:8063
    2. Settings -> Edit -> edit profile image (pencil icon)
    3. Upload a PDF or something that's not an image. You should get a red centered warning message. 
4. Try running `docker compose run --rm watch yarn run lint-check` to see linting run on all workspaces. Other commands work similarly.

#### Where should the reviewer start?
I would begin with `frontends/README.md`. 

#### Other Comments
Apologies for (another) rather large PR. Most of the file changes are import changes in`frontends/open-discussions`, but there's still a lot here. Will endeavor for smaller subsequent PRs.
